source/_posts/2023-04-27-year-of-the-voice-chapter-2.markdown

```diff
@@ -0,0 +1,158 @@
+
+[mm-algo]: https://github.com/jaywalnut310/vits/
+[piper-samples]: https://rhasspy.github.io/piper-samples
+[open-audio]: http://www.openslr.org/
+[audiobook]: https://librivox.org/
+[contact]: mailto:voice@nabucasa.com
+
+### Local speech-to-text with OpenAI Whisper
+
+[Whisper] is an open source speech-to-text model created by OpenAI that runs locally. Since its release in 2022, Whisper has been improved by the open source community to run on less powerful hardware by projects such as [whisper.cpp] and [faster-whisper]. In less than a year of progress, Whisper is now capable of providing speech-to-text for [dozens of languages][whisper-lang] on small servers and single-board computers!
+
+An {% my supervisor_addon addon="core_whisper" title="add-on using faster-whisper" %} is available now for Home Assistant. On a Raspberry Pi 4, voice commands can take around 7 seconds to process with about 200 MB of RAM used. An Intel Core i5 CPU or better is capable of sub-second response times and can run larger (and more accurate) versions of Whisper.
+
+[Whisper]: https://github.com/openai/whisper
+[whisper-lang]: https://github.com/openai/whisper#available-models-and-languages
+[whisper.cpp]: https://github.com/ggerganov/whisper.cpp
+[faster-whisper]: https://github.com/guillaumekln/faster-whisper/
+
+## Wyoming: the voice assistant glue
+
+Voice assistants share many common functions, such as speech-to-text, intent-recognition, and text-to-speech. We created the [Wyoming protocol][Wyoming] to provide a small set of standard messages for talking to voice assistant services, including the ability to stream audio.
+
+Wyoming allows developers to focus on the core of a voice service, without having to commit to a specific networking stack like HTTP or MQTT. This protocol is compatible with the upcoming [version 3.0 of Rhasspy][Rhasspy], so both projects will be able to share voice services.
+
+With Wyoming, we’re trying to kickstart a more interoperable open voice ecosystem that makes it easy to share components across projects and platforms. Developers and scientists wishing to experiment with new voice technologies need only implement a small set of messages to integrate with other voice assistant projects.
+
+The Whisper and Piper add-ons mentioned above are integrated in Home Assistant via the new [Wyoming integration]. Wyoming services can also be run on other machines and still integrate into Home Assistant.
+
+[Wyoming]: https://github.com/rhasspy/rhasspy3/blob/master/docs/wyoming.md
+[Rhasspy]: https://github.com/rhasspy/rhasspy3/
+[Wyoming integration]: https://next.home-assistant.io/integrations/wyoming/
+
+## ESPHome powered voice assistants
+
+[ESPHome] is our software for microcontrollers. Instead of programming, users define in a YAML file how their sensors are connected. ESPHome will read this file and generate and install software on your microcontroller to make this data accessible in Home Assistant.
```

Comment:
```suggestion
[ESPHome] is our software for microcontrollers. Instead of programming, users define how their sensors are connected in a YAML file. ESPHome will read this file and generate and install software on your microcontroller to make this data accessible in Home Assistant.
```